products/views.py

```diff
@@ -1,3 +1,74 @@
 from django.shortcuts import render
 
-# Create your views here.
+import json
+
+from django.views              import View
+from django.http               import JsonResponse
+from django.core               import serializers
+from products.models           import Menu, Category, Product, ProductImage, Ingredient, Allergen, AllergenStatus, ProductAllergen
+
+class ProductView(View):
+    def get(self, request):
+        category = request.GET.get('category')
+        limit    = int(request.GET.get('limit', 30))
+        offset   = int(request.GET.get('offset', 0))
+
+        if not category:
+            return JsonResponse({'message':'NONE_CATEGORY'}, status=401)
+
+        categories = Category.objects.filter(name=category)
+        products   = Product.objects.filter(category__name = category, relative_product = None)[offset:offset+limit]
+
+        results = [
+            {
+                "category" : [{
+                    "id"   : category.id,
+                    "name" : category.name,
+                } for category in categories],
+
+                "products" : [{
+                    "id"    : product.id,
+                    "name"  : product.name,
+                    "image" : [product_image.image_url for product_image in ProductImage.objects.filter(product=product)],
```

Comment:
`suggestion`과 같이 역참조 관계를 이용해서 작성하는법을 연습하시면 좋을 것 같습니다. 
```suggestion
                    "image" : [ product_image.image_url for product_image in product.productimage_set.all() ],
```